### ISSUE TITLE
`/subscriptions/sites`: Display all sites user is subscribed to - use mock data.

### ISSUE THREAD

**Issue by ivan-ottinger** [Automattic/wp-calypso#74966]
The goal of this task is to introduce the following list of sites the user is subscribed to (using mock data - as the API is not ready yet):

![Markup on 2023-03-27 at 17:00:12](https://user-images.githubusercontent.com/25105483/227980554-82fd10f8-4c8d-49a7-8619-9e616b9972f6.png)

## Acceptance Criteria

**✅ Changes the task covers**

Looking at the design (inDLaEQV8jJ21O4WXawIsJ-fi-712-15534), display the list of all sites the user is subscribed to, including the following columns:

- Subscribed site
  - site icon (circular mask should be applied; we need to handle situation when there's no icon)
  - site title (we need to handle situation when there's no title; long title should be addressed with ellipsis)
  - site URL (site URL should be properly formatted - as can be seen in the design;  long URL should be addressed with ellipsis)
- Since (the date needs to be properly formatted - maybe respecting user's locale if possible) 
- Email Frequency

Please use mock data. Focus on the desktop view and check the RTL version.

What should happen when user clicks on the site title, site icon or the URL is currently being discussed.

**❌ Changes this task does not cover**

Some of the changes will be handled in separate tasks that are linked below.

- "New Posts" column
- "New Comments" column
- "..." (three dots) that will later open a pop-up: https://github.com/Automattic/wp-calypso/issues/74973
- Mobile view: https://github.com/Automattic/wp-calypso/issues/74968
- Retrieving and displaying user's real data: https://github.com/Automattic/wp-calypso/issues/75068
- Blue badge / tag for P2 sites: https://github.com/Automattic/wp-calypso/issues/75078

We are neither implementing the filter, search, nor site type drop-down as a part of this task.

**Comment by yansern** [comment]
> "..." (three dots) that will later open a pop-up

Just to clarify (to myself) that this issue does not cover the dropdown because it will be covered in a separate issue.

**Comment by ivan-ottinger** [comment]
> Just to clarify (to myself) that this issue does not cover the dropdown because it will be covered in a separate issue.

I have clarified the task description and linked to the relevant task that will cover the drop-down menu.

**Comment by mashikag** [comment]
If we are to go by the acceptance criteria, then the PR (https://github.com/Automattic/wp-calypso/pull/75014) should not fully resolve this task. The sites a user is subscribed-to are not being displayed yet. Only a sites list  component, populated with mock data, was created.

To be clear, I do see that the issue mentions the following:
> If the data are not yet available, feel free to start with mock ones.

However, it explicitly says to "feel free to start with mock ones" which I read as: start with mock data, but finish the issue with real data, once it is available. Am I interpreting this right, @ivan-ottinger ? If so the issue should be re-opened.

**Comment by mashikag** [comment]
On top of the above, I was late to notice that the PR (https://github.com/Automattic/wp-calypso/pull/75014) does not make the site logos oval, and does not include labels such as P2. Therefore, I am reopening the issue.

cc @JuanLucha

**Comment by mashikag** [comment]
I am just clarifying my above comment. 

> does not make the site logos oval

The current implementation does not apply a circular mask to the mock logo image. See how in the designs all the logos are displayed as circular image:
![Screen Shot 2023-03-29 at 13 41 53](https://user-images.githubusercontent.com/2019970/228525989-d8adfa4b-d101-49ed-9693-aa4fa19bdc4c.png)

> does not include labels such as P2

This is the label in our design:
![Screen Shot 2023-03-29 at 13 45 13](https://user-images.githubusercontent.com/2019970/228525817-e07c2c95-ce26-4353-a040-dccc5b85f578.png)

____
When writing this comment, I realized that we are missing a few other things. I will try to summarize all that we are missing below.
1. **We should handle long text.**
 - whether it is a long site name, long URL, or too long content for any other cell
 - we should never wrap text in a cell. We should use ellipsis instead (I already discussed this with @saygunnyc sometime ago)
2. **We should handle site logo, site name and site URL click.**
  - I am not 100% clear as to what should happen on each, but my guess would be to take the user to the site. 
  - @saygunnyc would you mind clarifying what should happen on a click at different parts of the site row? Also, what kind of mouse pointer would you expect when hovering over cells that are not clickable (if there are such cells)? 
3. **Only the site's domain name should be displayed.**
  - @JuanLucha you are right to assume that we will probably receive a site URL together with transfer protocol prepended, and possible forward slash appended (@TimBroddin may know exactly what we should expect). However, the URL should be treated before it is displayed so that only the domain name is displayed. As per Figma designs. 
4. **Display the date with the expected date format applied.**
  - The current implementation for displaying a date is simply the following: `date.toDateString()`. This neither displays a localized date format, nor the date format that is in our designs (e.g.: October 23, 2004). I am not sure myself what is the desired date format when we take all the supported locales/languages into consideration. Maybe @yansern, or @saygunnyc, can provide you with the expectations.
5. **Display the P2 label when a site is a P2 site.**
6. **Display site logos with circular mask applied.**
7. **Handle a site with no logo**
- @TimBroddin what happens when a site has no custom icon set? Will some default logo image URL be provided by the backend? Or this should be handled on the frontend?
8. **Handle a site with no title**
- From the experience when working with emails I believe sites may have undefined names. @TimBroddin will backend provide some default site name in such case? Or this should be handled on the frontend?

**Comment by TimBroddin** [comment]
> @JuanLucha you are right to assume that we will probably receive a site URL together with transfer protocol prepended, and possible forward slash appended (@TimBroddin may know exactly what we should expect). However, the URL should be treated before it is displayed so that only the domain name is displayed. As per Figma designs.

The site URL will be returned as https://domain.com (the blog's siteurl). As far as I can see, this won't include a trailer slash. 

> @TimBroddin what happens when a site has no custom icon set? Will some default logo image URL be provided by the backend? Or this should be handled on the frontend?

`site_icon` will be null when it isn't set.

> From the experience when working with emails I believe sites may have undefined names. @TimBroddin will backend provide some default site name in such case? Or this should be handled on the frontend?

I haven't seen any sites without title, but I can see no default being set in the code. This will be `null` as well. 

FYI, the patch (D106264-code) that adds external user support to the API is waiting on review by the mobile team. After they run their tests and approve it, I can deploy it. After deploying that task, [this PR](https://github.com/Automattic/wp-calypso/pull/75038) will make it into trunk fast.

**Comment by ivan-ottinger** [comment]
> However, it explicitly says to "feel free to start with mock ones" which I read as: start with mock data, but finish the issue with real data, once it is available. Am I interpreting this right, @ivan-ottinger ? If so the issue should be re-opened.

Thanks for the ping, Maciej!

I could have been more clear in the task description. When I was writing it, it wasn't clear whether the API and the query callback will be ready when the task is picked up.

Anyways, I am planning to pull in the current data in a separate task: https://github.com/Automattic/wp-calypso/issues/75068.

**Comment by ivan-ottinger** [comment]
Alright, I have clarified the task description and added the details there.

**Quick summary:**
- this task will use just the mock data - so we can focus on the actual display of the list before the API is ready
- other points brought up by @mashikag are now in the task description; thanks Maciej for brining them up! 🙂  
- since not all the details are clear yet (e.g. what should happen when somebody clicks on the Site icon), we can leave the discussion in this thread;
- changes we are still missing (e.g. applying the circular mask to the Site icon) can be considered as a part of this task - or -  can be also extracted out to their own tasks if preferred. 🙂  
- the P2 tag feature has its own task now (as I suspect this will need a bit more work): https://github.com/Automattic/wp-calypso/issues/75078
- retrieving and displaying user's real data has its own task now as well: https://github.com/Automattic/wp-calypso/issues/75068
- marking this task as `In progress`.

Related discussion: p1680096392295819-slack-C02TCEHP3HA

**Comment by JuanLucha** [comment]
Very nice discussion. Thanks everyone involved 🙇🏻‍♂️
I'll work on the points from @mashikag as part of this ticket.

**Comment by saygunnyc** [comment]
> We should handle site logo, site name and site URL click.
I am not 100% clear as to what should happen on each, but my guess would be to take the user to the site.
@saygunnyc would you mind clarifying what should happen on a click at different parts of the site row? Also, what kind of mouse pointer would you expect when hovering over cells that are not clickable (if there are such cells)?

Let's follow the experience on https://wordpress.com/sites.  _– Underline text links, mouse pointer | Form elements, beam_ All (icon, name, URL) should be linked to the site in a new window.

> Maybe @yansern, or @saygunnyc, can provide you with the expectations. 

The format currently introduced on the designs, `September 12, 2022` is friendlier than `9/12/2022`. To make it even friendlier, we can follow https://wordpress.com/sites, –hours, days, followed by date. 

**Since**
2 hours
2 days
March 15, 2023 

Let me know your thoughts!

**Comment by yansern** [comment]
Thanks @saygunnyc. We'll go with how it's implemented on Sites list.
